### PR TITLE
Passback updates

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Updates the passback listener with:

- new value for the `passback` targeting parameter based on the breakpoint e.g. `teadsdesktop` or `teadsmobile`
- early returns if the required iframe or elements are not found
- reorder `slotRenderEnded` listener registration
- wrap dom mutations with fastdom
- readability and logging improvements

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by chrislomaxjones — APPROVED** [review]
Thanks for including the clear documentation

File: static/src/javascripts/projects/commercial/modules/messenger/passback.ts [hunk 1]
```diff
@@ -25,6 +20,17 @@ const outstreamMobile: [number, number] = [
 	adSizes.outstreamMobile.height,
 ];
 
+const getValuesForKeys = (
+	keys: string[],
+	valueFn: (key: string) => string[],
+): Array<[string, string[]]> => keys.map((key) => [key, valueFn(key)]);
+
+const getPassbackValue = (source: string): string => {
+	const isMobile = (getBreakpoint() as string).startsWith('mobile');
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/messenger/passback.ts:29]: `getBreakpoint` from detect-viewport is typed!

Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/messenger/passback.ts]: Thanks have updated :)

**Review by Jakeii — APPROVED** [review]

**Review by chrislomaxjones — APPROVED** [review]
Looks great!

File: static/src/javascripts/projects/commercial/modules/messenger/passback.ts [hunk 2]
```diff
@@ -42,174 +48,216 @@ const init = (register: RegisterListener): void => {
...
-						['slot', ['inline1']],
-					];
-
-					/**
-					 * Create a new passback ad slot element
-					 */
+			/**
+			 * Create a new passback ad slot element
+			 */
+			const createNewSlotElementPromise = updateInitialSlotPromise.then(
```
[the remark this answers is not in the document]
Comment (Jakeii) [reply static/src/javascripts/projects/commercial/modules/messenger/passback.ts:110]: I don't think it'd matter though right? The function is still called async or not

**Pull request merged**